contrib/userscript/unnoticed.user.js

```diff
@@ -0,0 +1,201 @@
+// ==UserScript==
+// @name         Unnoticed Leaderboards
+// @namespace    https://github.com/christopher-dG/unnoticed
+// @version      0.1
+// @description  Display unranked leaderboard entries gathered by Unnoticed on their respective beatmap pages
+// @author       Node
+// @match        https://osu.ppy.sh/*/*
+// @include      https://p9bztcmks6.execute-api.us-east-1.amazonaws.com/*
+// @grant        GM_xmlhttpRequest
+// @require      https://code.jquery.com/jquery-3.2.1.min.js
+// @require      http://momentjs.com/downloads/moment.min.js
+// @connect      p9bztcmks6.execute-api.us-east-1.amazonaws.com
+// @run-at document-end
+// ==/UserScript==
+
+(function(){
+    'use strict';
+
+    var mods_enum = {
+        ''   : 0,
+        'NF' : 1,
+        'EZ' : 2,
+        ''   : 4,
+        'HD' : 8,
+        'HR' : 16,
+        'SD' : 32,
+        'DT' : 64,
+        'RX' : 128,
+        'HT' : 256,
+        'NC' : 512,
+        'FL' : 1024,
+        'AT' : 2048,
+        'SO' : 4096,
+        'AP' : 8192,
+        'PF' :16384 
+    };
+
+    function accuracy(c300, c100, c50, cmiss){
+        return +
+            ((c300 * 300 + c100 * 100 + c50 * 50)
+            /  (c300 * 300 + c100 * 300 + c50 * 300 + cmiss * 300)
+            *  100)
+            .toFixed(2);
+    }
+
+    function grade(c300, c100, c50, cmiss, mods){
+        var ctotal = c300 + c100 + c50 + cmiss;
+        var p300 = c300 / ctotal;
+        if(c100 == 0 && c50 == 0 && cmiss == 0){
+            if(mods.includes("HD"))
+                return "XH";
+            else
+                return "X";
+        }else if(p300 > 0.9 && c50 / ctotal < 0.01 && cmiss == 0){
+            if(mods.includes("HD"))
+                return "SH";
+            else
+                return "S";
+        }else if(p300 > 0.8 && cmiss == 0 || p300 > 0.9){
+            return "A";
+        }else if(p300 > 0.7 && cmiss == 0 || p300 > 0.8){
+            return "B";
+        }else if(c300 > 0.6){
+            return "C";
+        }else{
+            return "D";
+        }
+    }
+
+    function mods(enabled_mods){
+        var return_array = [];
+        for(var mod in mods_enum){
+            if((mods_enum[mod] & enabled_mods) != 0)
+                return_array.push(mod);
+        }
+        return return_array;
+    }
+
+    function mods_string(enabled_mods){
+        if(mods(enabled_mods).length > 0)
```

Comment:
Not that `enabled_mods` is a very expensive function, but store the mods in a variable and reuse it.